app/src/main/java/com/lastaoutdoor/lasta/data/db/ActivitiesRepositoryImpl.kt

```diff
@@ -7,69 +7,117 @@ import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.firestore.SetOptions
 import com.lastaoutdoor.lasta.R
 import com.lastaoutdoor.lasta.data.model.profile.ActivitiesDatabaseType
+import com.lastaoutdoor.lasta.data.model.user.UserModel
 import com.lastaoutdoor.lasta.repository.ActivitiesRepository
 import javax.inject.Inject
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
 
+/**
+ * Implementation of the ActivitiesRepository interface. This class provides methods to interact
+ * with the Firebase Firestore database.
+ *
+ * @property database The Firebase Firestore database instance.
+ * @property context The application context.
+ */
 class ActivitiesRepositoryImpl
 @Inject
 constructor(private val database: FirebaseFirestore, private val context: Context) :
     ActivitiesRepository {
   private val activityConverter = ActivityConverter()
 
+  /**
+   * Adds a new user to the activities database.
+   *
+   * @param user The FirebaseUser to add to the database.
+   */
   private fun addUserToActivitiesDatabase(user: FirebaseUser) {
-    val userDocumentRef =
-        database.collection(context.getString(R.string.activities_database_name)).document(user.uid)
-    val userData =
-        hashMapOf(
-            "Hiking" to arrayListOf<ActivitiesDatabaseType.Trail>(),
-            "Climbing" to arrayListOf<ActivitiesDatabaseType.Climb>())
-    userDocumentRef.set(userData, SetOptions.merge())
+    try {
```

Comment:
Nice ! we should definitively check the rest of the project for potentials errors thrown. 

A snackbar with a user friendly error is also a thing to have in the app soon.